### DISCUSSION TITLE
유저테스트 피드백 정리

### DISCUSSION THREAD

**Discussion by Jaymyong66** [woowacourse-teams/2024-code-zap#432] [QA]
# 리브
###  FE
- (로그인) 로그인이 안된 상태에서 '새 템플릿' 버튼이나 '내 템플릿' 네비를 누르면 아무 피드백이 없다.버튼을 누르면 토스트 메시지를 띄울까? 

  - 추후 비회원도 우리 서비스 중 사용할 수 있는 기능이 있으면 좋겠다. Explore 같은. 이 기능을 보고 해당 비회원도 회원이 되고 싶게 하면 좋겠다.(마위)

- (태그) 태그를 쓰는데 태그 칸이 너무 작고 잘 안 보인다. 있는 줄도 몰랐다. 또 태그 삭제 버튼이 너무 작아서 리브가 거북목이 되었다.
  - 추가로 태그에 띄어쓰기를 했다가 띄어쓰기하면 태그 입력이 종료됨을 자연스럽게 알아차렸다. 또 태그를 이모지로 만들기도 하더라.

- (태그) 태그 넓이가 고정이라 태그의 글자수가 조금만 길어져도 줄바꿈이 되고 있다. 이를 wrap으로 하거나 최대 넓이를 늘려야할 것 같다.

- (내 템플릿 페이지) 날짜가 길어져서 줄바꿈이 되며 깨진다

- (내 템플릿 페이지) 페이징 버튼이 너무 작다

- (업로드) 스니펫에 확장자 명을 안써도 하이라이팅이 된다. yaml 하이라이팅도 해줬으면 좋겠다 -> 자스로 박아둬서 그렇다!

- (업로드) 제목과 설명의 입력란의 큰 차이를 모르겠다

- (카테고리) 카테고리도 삭제하고 싶다.

### BE
- 뭐가 있지..?

### 있었으면 하는 기능
- 진짜 자주 쓰는 것이 위로 가거나, 아예 핀을 꽂고 싶다.(즐겨찾기?)
- 파일명 뒤에 확장자 쓰는게 번거롭긴 하지만 괜찮다. 확장자를 쓰거나 하이라이팅을 위한 드롭다운도 같이 있으면 좋을듯?

### 긍정적인 부분
한눈에 어디에 뭐가 있는지 딱 보여서 편리했다!

[회원가입 페이지]
- 비밀번호가 영문자+숫자의 8글자 이상인 것을 안내 메시지에서 잘 확인하고 적용함.

[업로드 페이지]
- 카테고리 잘 생성하고 타이틀과 설명 잘 입력함.
- 파일명 진입과 확장자까지 잘 씀.
- 지정된 task

[내 템플릿 페이지]
- 템플릿을 찾으라는 태스크에선 바로 검색을 통해 찾음. 이때 템플릿의 description을 주로 봤다.
- 복사 버튼도 잘 씀 !
- 삭제, 태그만 보고 싶을 때도 직관적으로 잘 찾음.

[로그아웃]
- 간단해서 좋다! 

---

# 구구

### 기획
- 탈퇴가 안된다.
- 스니펫 내용을 입력해주세요 -> 스니펫이 무엇인지 헷갈려한다. 
  - 어떨 때는 파일, 어떨 때는 스니펫. 소스코드는 어떤가?
- 스니펫 작성 시, 내가 몇 자를 입력했는지 그때그떄 피드백이 가도 좋을 것 같다.
- 생성 수정 시간은 초 단위로. 38분에 만들었는데 36분으로 나온다.
- 템플릿을 지워도 태그는 살아남는다. 태그에 해당하는 템플릿이 존재하지 않으면 태그도 삭제해라.
- 카테고리 선택 부분을 이제 카테고리 없음으로만 제공해도 되지 않나...(?)
- 태그 입력시 #를 당연하게 생각해서 입력했다.

### FE
- (리브와 중복) 비로그인 상태에서 새 템플릿 눌렀을때 아무 피드백 없는 것.
- 태그가 잘 안보인다. 태그 어디에 추가하는 거지? 설명에 태그 추가하려함
- 태그 삭제 버튼 부분이 너무 작다

- (인프라) www 붙이는거 불편하다
- 400 에러 발생시 아무 피드백이 없다.

- (전체적인 Input) 커서의 위치가 조금 어색하다. input이 아닌 padding 부분을 클릭하면 input focusing이 되지 않아서 그런 것 같다. 다만 border-radius를 고려하면 input의 위 아래 padding만 없애보면 어떨까..

- (카테고리) 카테고리 생성할 때 무조건 엔터를 입력해야하는 것이 불편하다. 뒤로가기 버튼이 있는 것이 좋은가?
- (카테고리) 카테고리가 선택사항인지 명확하지 않다.
- (카테고리) 카테고리 글자 오버시 어떻게 처리중?

- (태그) 태그의 내용물과 삭제 버튼 간의 얼라인이 맞지 않는다. 또 태그 간 gap을 늘려 달라
- 검색과 정렬 드롭다운 간 얼라인도 미세하게 맞지 않는다. 

### BE
- Response Header의 Remote Address에서 IP 주소가 보이면 안된다. (nginx 설정)
- 에러 상황에는 하나의 에러 상황만 표시해라 (여러개를 입력하지 마라 - ex: 템플릿 제목은 어쩌고, 파일 이름은 어쩌고,,,) -> 아마도 백앤드에 대한 피드백?
- 스니펫 입력 중 많은 것을 입력했을 때, 에러 메시지로 xxx바이트를 초과했습니다로 온다. -> 글자수로 피드백을 줘라
- 서버에 저장 안둬면 토글 떠여함

**Comment by Jaymyong66** [comment]
### 기타 의견
[마위]
  - (스니펫) 업로드, 템플릿 수정 모두 스니펫 입력의 높이가 너무 작다. 현재 약 10줄로 고정되어있음.

- (내 템플릿 페이지) 템플릿 설명이 길어지면 wrap해야할듯.  padding이나 gap도 더 줘야할듯. (제목과 설명 행간을 늘리고 싶음)
 
  - (로그인) 로그인 실패 시, 단순히 '로그인이 실패하였습니다' 피드백만 준다. 왜 실패했는지에 대한 원인을 몰라서 여러번 로그인을 시도함. 이메일이 실패했는지, 비밀번호가 실패했는지 구체적으로 알려줘야할까?

    - 이에 대해 이메일 유효성을 검증해야하나? 라는 의구심이 생겼다. 실제로 쓰지 않는 아무 이메일을 입력하다보니 로그인 실패 원인을 두 군데에서 찾는 듯 했다.
    - 다른 개선 방향으로 이메일 -> 아이디로 변경하자는 의견도 있었다. [아이디 찾기] 기능이 생긴다면 [이메일 찾기]는 좀 이상할거 같음

- (템플릿 업로드) Add Snippet 버튼 한글화..

- (템플릿 업로드) 스니펫을 여러 개 입력할 때, 스니펫 추가 버튼을 누르면 다음 스니펫의 filename 입력창으로 포커싱 되면 좋지 않을까?

- (로그아웃) 로그아웃 버튼 누르면 확인 모달 띄워야하나? 너무 바로 로그아웃 되는 것 같기도.

- (태그) 템플릿을 찾으라는 task에서 태그가 좌측 하단이라 잘 보이지 않았고 검색을 할 생각밖에 들지 않았다고 했다. 태그를 검색창 하단에 놓는다면?


- 스니펫을 입력할 때 인텔리제이에서 복붙해옴. 생각해보면 사용자가 에디터에서 입력을 잘 하진 않을듯?

**Reply by Jaymyong66** [reply]
자동 로그인 체크 박스...? 혹은 브라우저의 아이디 비밀번호 기억 기능이 잘 동작하도록 개선?

**Reply by Jaymyong66** [reply]
카테고리 생성 후 방금 생성한 것 자동 선택되도록

**Reply by Hain-tain** [reply]
- [헤인의 개인적 생각] 카테고리를 새로 만드는 것은 생각보다 직관적으로 빠르게 잘 습득하신 것 같아 좋은 것 같다. 다만, 카테고리 생성 후 자동으로 선택되게 하는 것, 이미 존재하는 것을 적고 엔터 쳤을 때 해당 카테고리를 선택하게 하는 기능이 추가되면 좋을 것 같다는 생각이 들었다.

**Comment by zeus6768** [comment]
구구 피드백 추가합니다. 

- 파일명 입력 칸이 있는 줄 몰랐다. 코드 입력하는 칸과 더 뚜렷하게 구분해달라.

**Comment by Hain-tain** [comment]
### 카피
(인터뷰)
- 체계적으로 저장하진 않지만, 북마크를 해두는 편이다. 로그인같은 경우 중복되니까 다른 사람들이 작성한 좋은거 있으면 북마크 해두는 편.
- 북마크 하는 이유=> 클릭 한 번으로 되니까 간편하기 때문! 코드 복붙 하기 귀찮아서 북마크로 다시 찾아보게 된다.
- 찾을때 불편하진 않았다. 사람마다 다를테지만, 많이 저장하지 않는 편이라서 양 자체가 부담이 되진 않았다.
- 재사용하고 싶은 코드를 보면 제일 먼저 떠오르는 생각? 어디보자~ 하면서 자세히 살펴보고, 똑같이 적용해도 안될때가 있으면 당황하고 그런 것 같다.
- 어떤 코드들을 재사용하고 싶나요? 아무래도 반복되는 코드들 재사용하고 싶다. 작성하는 사람의 아이디어가 필요한게 아니라 설정이라든지 정형화된 내용들을 재사용하면 좋을 것 같다.
- 재사용하고 싶은 코드의 파일의 크기는 어땠나? 여러개로 나눠져 있지만 연관관계가 다 있는 것 같다. 하나하나 참고하게 된다. 
- 내가 코드를 저장해둔다면, 어떤 기능이 있으면 빠르게 인출할 수 있을까? => 소제목이 있으면 좋을 것 같다.옆에서 탭으로 바로바로 갈 수 있는 기능이 있으면 편할 것 같다. 본인이 커스텀하는거면 설명같은걸 메모할 수 있으면 좋을 것 같다.
- 이런 서비스는 로그인 해야할 것 같은 생각이 들어서 회원가입은 큰 장벽은 아니다. 그러나 코드가 길어지거나 많이지면 보기 힘들지 않지 않을까? 사용해보지 않아서 모르겠다.
- 저장을 할 때 북마크보다는 시간이 걸리는데, 이런 수고스러움을 충분히 이겨낼만할까요? 북마크를 해도 결국 한계가 있는 것 같아요. 알아보기도 쉽지 않고, 페이지만 이동하다보니, 관리도 쉽지 않고. 그래서 프로그래밍을 업으로 삼는다면 비슷한 코드를 계속 마주치게 될텐데 이런게 있으면 좋을 것 같아요.
- 카테고리, 태그 있으면 좋을 것 같나요? 좋을 것 같아요. 아무래도 각자만의 분류 기준이 있을거니까. 카테고리 지정해주면 범용성이 있어야 할 것 같고, 커스텀 가능하다면 큰 어려움은 없을 것 같아요. 분류에 도움은 될 것 같아요.
- 이외에 유용한 기능 생각나는게 있다면? 1차적으로 코드를 저장하는거니까 그거에 대한 플로우가 편해야할 것 같고, 간편하게 몇가지만 넣으면 바로 저장할 수 있으면 좋을 것 같고, 다른 사람들이 뭘 저장하는지 볼 수 있으면 좋을 것 같다.
- 쓰실건가요..? => 최근에 코드를 저장했던게 프리코스 시험이었는데 그때 알았다면 사용해볼만 했을 것 같다.

---
(QA)
- 카테고리 가이드 길이가 너무 길다. 
- 태그 칸이 분리가 잘 안되어있어서 잘 인지가 안되는 것 같다. 분리가 되면 좋을 것 같다.
- 예전보다 디자인이 너무 예뻐졌다.
- 태그 색상이 흐린 것 같다.
- 쓰는 것 자체는 크게 어렵지 않다. 기능적으로 불편함을 느끼진 않았다. 근데 뭔가 여백이 많아서 비어보이는 느낌이 든다. 
- 설명 태그도 칸이 잘 보이지 않는다.
- 코드가 한 눈에 잘 보이면 좋겠다. 지금 코드 뷰어가 좁아보인다.
- 모든 태그 버튼과 태그 사이의 정렬이 묘하게 안맞는 것 같다. 
---
내가 느낀점
- 확장자를 입력하는 것을 놓치는 것 같다.
- 파일명의 길이- 경고문 계속 떠서 불편함.
